docs/demo/test-api.js

```diff
@@ -69,6 +70,18 @@ module.exports = function (config) {
       return Promise.resolve(ids[id])
     },
     tagsAndCategories: () => Promise.resolve(config.tagsAndCategories),
+    settings: () => Promise.resolve(config.settings),
+    setSetting: (name, value, addOptions) => {
```

Comment:
This allows a single setting to add multiple items to the settings object.

For instance, to add spellcheck to a CodeMirror instance, you need too initiate it with the options `inputStyle='content-ediatble'` and `spellcheck=true`. As an api call, setting this is
```js
api.setSetting('spellcheck', true, {editor: {inputStyle: 'content-editable', spellcheck: true}})
```
which makes `_admin-config.yml` (or `api.settings()`):
```yaml
options:
   spellcheck: true
editor:
   inputStyle: 'content-editable'
   spellcheck: true
```

The `editor` settings are [passed to CodeMirror](https://github.com/jaredly/hexo-admin/pull/91/files#diff-93bc2a7742610064cd1e94a3138c37b1R28).

The settings checkbox takes enable and disable options and passes them appropriately to `setSettings`. The checkbox has
```yaml
enableOptions: {editor: {inputStyle: 'contenteditable', spellcheck: true}}
disableOptions: {editor: {inputStyle: null, spellcheck: false}}
```
So the above is set in `_admin-config.yml` or a call to `api.settings()` when the box is enabled, and when it's diabled it looks like
```yaml
options:
   spellcheck: false
editor:
   inputStyle: null
   spellcheck: false
```

Perhaps renaming `addOptions` to `settingsToAdd` would be better?